django/s3widget/models.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any, cast
+from uuid import uuid4
+
+from django.contrib.admin.widgets import AdminFileWidget
+from django.db.models.fields.files import FieldFile, FileField
+
+from .widgets import S3AdminFileInput, S3FakeFile, S3FormFileField
+
+
+class S3FieldFile(FieldFile):
+    def save(self, name, content, save=True):
+        if not isinstance(content, S3FakeFile):
+            return super().save(name, content, save)
```

Comment:
Sorry still trying to understand the relationships between these objects - under what circumstances will this line get executed?

Reply:
it is actually responsible for storing the temporary uploaded file to a persistent storage.